src/probe/probe.c

```diff
@@ -1004,21 +1042,35 @@ int probe_point_add(uint32_t count, struct probe_point *probe)
 		}
 
 		if (probe[i].purpose == PROBE_PURPOSE_EXTRACTION) {
-			/* check if buffer exists */
+#if CONFIG_IPC_MAJOR_4
+			dev = ipc_get_comp_by_id(ipc_get(), IPC4_COMP_ID(probe[i].buffer_id.fields.module_id,
```

Comment:
This is nice! I was worried we'd need more code to do the module based lookup, but it seems existing infra was pretty close already.

Reply:
I was worried too! But with some extra for loops we can quickly find our buffer